### PR TITLE
Support new NSSL cloud microphysics 3-moment option

### PR DESCRIPTION
**Pull request by MicroTed**
## Description

Adds namelist flag (nssl_3moment) and tracer variables to use NSSL 3-moment microphysics option (3-moment for rain, graupel, and hail)

### Issue(s) addressed

Supports PR for CCPP: https://github.com/ufs-community/ccpp-physics/pull/110

Also initializes the fscav_aero array to nulls so that the logfile.out doesn't have random memory values and look like a binary file (and sometimes screw up xterm settings with random control codes when the file is viewed with more/less)

## Testing

RT with ufs_model has been done on hera. Results change as expected for SDFs that use the NSSL scheme (FV3_rrfs_v1nssl and FV3_WoFS_v0). Also tested on MacOS (intel cpu;gfortran). Weather model branch with FV3/ccpp update is

https://github.com/MicroTed/ufs-weather-model/tree/feature-nssl3m
  - or:
url= https://github.com/MicroTed/ufs-weather-model.git
branch=feature-nssl3m

RT does not include the 3-moment option, but I'm not sure one is really needed.

Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
  Yes, it will change regional_wofs, rrfs_v1nssl, and rrfs_v1nssl_nohailnoccn because of some changes to the NSSL microphysics scheme that is used with the tests.
- Please commit the regression test log files in your ufs-weather-model branch
 /scratch1/BMC/gsd-fv3-dev/Ted.Mansell/ufs/ufs-nssl3m/tests/RegressionTests_hera.log.nssl3m

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
   url = https://github.com/MicroTed/ccpp-physics
   branch = ufsdev-nssl3m
Concurrent PR for CCPP: https://github.com/ufs-community/ccpp-physics/pull/110

### CODE REVIEW

File: ccpp/data/CCPP_typedefs.F90 [hunk 1]
```diff
@@ -1061,7 +1061,8 @@ subroutine gfs_interstitial_setup_tracers(Interstitial, Model)
              n /= Model%ntrw  .and. n /= Model%ntsw  .and. n /= Model%ntrnc   .and. &
              n /= Model%ntsnc .and. n /= Model%ntgl  .and. n /= Model%ntgnc   .and. &
              n /= Model%nthl  .and. n /= Model%nthnc .and. n /= Model%ntgv    .and. &
-             n /= Model%nthv  .and. n /= Model%ntccn .and. n /= Model%ntccna .and.  &
+             n /= Model%nthv  .and. n /= Model%ntccn .and. n /= Model%ntccna  .and.  &
```
Comment (BrianCurtis-NOAA) [inline ccpp/data/CCPP_typedefs.F90]: remove extra space before `&`

File: ccpp/data/GFS_typedefs.F90 [hunk 2]
```diff
@@ -3888,7 +3893,7 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
                                sedi_semi, decfl,                                            &
                                nssl_cccn, nssl_alphah, nssl_alphahl,                        &
                                nssl_alphar, nssl_ehw0, nssl_ehlw0,                    &
```
Comment (BrianCurtis-NOAA) [inline ccpp/data/GFS_typedefs.F90]: move `&` with others

**Review by BrianCurtis-NOAA — APPROVED** [review]

File: ccpp/data/CCPP_typedefs.F90 (hunk 1 above)
Comment (MicroTed) [reply ccpp/data/CCPP_typedefs.F90]: Sure, no problem

File: ccpp/data/GFS_typedefs.F90 (hunk 2 above)
Comment (MicroTed) [reply ccpp/data/GFS_typedefs.F90]: Done

File: atmos_cubed_sphere
Comment (DusanJovic-NOAA) [inline atmos_cubed_sphere:1]: Why is this submodule updated?

Comment (MicroTed) [reply atmos_cubed_sphere:1]: @DusanJovic-NOAA  Hmm, I don't know -- maybe I didn't do the merge correctly? I didn't intentionally update any files in there. I'll take a look.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by grantfirl** [comment]
@MicroTed https://github.com/ufs-community/ccpp-physics/pull/110 is merged. Please revert .gitmodules and update the ccpp/physics submodule pointer to the latest ufs-dev commit: https://github.com/ufs-community/ccpp-physics/commit/c751a5a6e1a5ae1f3f79e667a86e8dee0e62879f

**Comment by MicroTed** [comment]
@grantfirl  Thanks! OK, it should be updated now.

**Pull request merged**